GH-651 Review 2

Syther007 opened the pull request.

bertllll committed:
GH-580: test for data_directory_from_context with opt dir and implementation … (#248)

* test for data_directory_from_context with opt dir and implementation in fn

* fixed naming and optimizing call for add_chain_specific_directories()

* add_chain_specific_directories to tests in  bootstrapper.rs, mod.rs, node_configuraton_standard.rs

* calculate_configured_setup returns None -> panicked at 'called  on a  value'

* formatting

* cleaning misstake

* commit changes - move to laptop

* fix tests failing on the chain specific data direcotry

* changes based on Berts review comments

* formatting

* remove todo causing crash

* half way to fix cleanup_after_deceased_clients_integration

* failing on creating log file in node_logs_panic_integration

* change joins for add_chain_specific_directories fn

* remove unused import

* add_chain_specific_directories moved to masq_lib::utils, introduced new function node_chain_specific_directory for multinode tests

* remove comments

* remove comments, replace joins wtih add_chain_specific_directories fn, test for chain specific directories uncompleted but passing

* Revert "remove unused import"

This reverts commit c0f205a2cd6c5ecb1d53e48cf2763ffacd1455f4.

* pull and resolve conflicts

* vec! with foreach for test server_initializer_collected_params_senses_when_user_specifies_data_directory_without_chain_specific_directory

* add more scenarios for test with custom data_directory_opt in test server_initializer_collected_params_senses_when_user_specifies_data_directory_without_chain_specific_directory

* continue of solving Bert's comments, start of modifying UI to show full path for user specified data-directory

* try to retrieve chain-name from inner.values

* chain_name collecting with map().collect, value_value collecting with match

* move match for value_value to separate fn to clar dump_setup

* chainge test for right fn calls, added message about changed data-directory to match chain parameter

* change of chain_specific_data_directory to add alway chain_specific_dir to path set by user, so chain dir could be duplicated, finalize test for masq UI to check various UiSetupResponsValueStatus

* new integration test for specific data_directory ensure_data_directory_has_specific_chain_direcotry_within

* fix Default arm for directory path enrichment with chain specific directory

* integration test to ensure setup command do the right job to add chain specific directory

* change data-directory, comparsion and panic messages in chain_specific_directory_test_integration test, remove comments inn setup_command

* fix formatting for expected in chain_specific_directory test, remove unused import from setup_command

* finalize test for setup_reporter calculate_fundamentals_with_chain_specific_dir

* commented code to test in linux

* compilable version to test in linux

* move changes to desktop

* for bert from my linux machine

* GH-580: showing the right direction

* fixing more tests

* GH-580: ongoing repairment in setup_reporter

* GH-580: fixed data directory niches that were causing a lot of troubles

* remove unvanted println

* formatting

* fixed test calculate_setup_with_chain_specific_dir to pass on mac, resolved last Berts comments

* remove unused imports and chain from mock node

* remove unused code from setup_command

* resolving review comments - add closure for retriving data-dir and chain

* resolving comments: cleaning setup_command tests, cleaning setup_reporter:get_modified_setup, fix misspe<lls

* try to send less data to determine_setup_value_of_set_data_directory fn

* remove unused data from SetupReporterReal::determine_setup_value_of_set_data_directory

* formatting

* fixed test for get_modified_setup::get_modified_setup_user_specified_data_directory_depends_on_new_chain_on_success

* resolved review comments on ensure_data_directory_has_specific_chain_directory_within_integration test

* split of test calculate_setup_with_chain_specific_dir into two separate tests

* resolving review comments, renaming, optimization, fixing tests

* resolving review comments - mainly renaming

* Half way to get rid of DirsWrapper from go fn in DumpConfigRunnerReal

* remove DirsWrapper from arguments to DumpConfigRunner for DumpConfigRunnerReal

* fixed windows tests for determine_fundamentals

* renaming functions, removing TODOs and removing unnecesary attributes from test

* fixed node_chain_specific_data_directory and impl MASQRealNode plus try to fix node_creates_log_file_with_heading_integration - continue next

* fixed node_creates_log_file_with_heading_integration

* remove unused import and debug print, tests failing in ci/all.sh but passing when run from cmd line

* fix paths in tests for windows: setup_reporter

* fix windows paths for test server_initializer_collected_params_senses_when_user_specifies_data_directory_without_chain_specific_directory

* fix setup_command and broadcast_handler tests: added missing data-directory and chain parameters to config

* setup_command removign ropsten and correcting UiResponseValue, broadcast_handler: removing unwanted piece of code, shared_schema: DATA_DIRECTORY_HELP, setup_reporter: fix test

* correction of DATA_DIRECTORY_HELP

* lazy_static for STANDARD_ARG_LIST in shared_schema

* separate help for --data-directory in Node, Config Dump and UI

* remove duplication of data-directory clap app argument, add dirs to masq_lib to compute static constant DATA_DIRECTORY_DAEMON_HELP, fix broadcast_hadnler test

* move data-directory back to shared schema::shared_app::head

* remove typo misstake

* apply Dans comments on review 6, put code in the order, formatting and moving code thru directories and files

* remove unused import

* resolve DATA_DIRECTORY_DAEMON_HELP for masq UI

---------

Co-authored-by: Vojtěch Parkán <vojtechparkan@Vojtechs-iMac.home>
Co-authored-by: Vojtěch Parkán <vojtechparkan@Vojtechs-iMac.local>
Co-authored-by: Vojtěch Parkán <czarte@gmail.com>
Co-authored-by: Bert <Bert@Bert.com>

bertllll committed:
Little renamings (#333)

* little renaiming

* little renaming: formatting

---------

Co-authored-by: Bert <Bert@Bert.com>

utkarshg6 committed:
GH-745: Macro for creating HashMap (#337)

* macros-experiment: add hashmap!()

* GH-745: gitignore files

* GH-745: untrack file

* GH-745: trigger actions

Syther007 committed:
GH651: Improved TryTransmitToHopperArgs::New

Syther007 committed:
GH-651: added new test new_http_request_creates_new_exhausted_entry_inside_dns_retries_hashmap_zero_hop

Syther007 committed:
GH-651: Simple Renaming

Syther007 committed:
GH-651: Added condition to test: handle_dns_resolve_failure_sends_message_to_neighborhood

utkarshg6 committed:
GH-692:  Allow UI to configure --min-hops during runtime (#316)

* GH-692: ignore file dbnavigator.xml

* GH-692: add tests for command execution of set-configuration

* GH-692: revert the version number

* GH-692: add test for the error case

* GH-692: improve tests to work on macos

* GH-692: improve tests to work on macos

* GH-692: ignore dbnavigator.xml

* GH-692: remove the redundant gitignore line

* GH-692: add tests for setting min_hops using set-configuration

* GH-692: add TODOs

* GH-692: migrate the Consuming Wallet change and Password Change to a new SetConfigurationMessage

* GH-692: update field names inside SetConfigurationMessage and variants name inside ConfigurationChange

* GH-692: update field names inside SetConfigurationMessage and variants name inside ConfigurationChange

* GH-692: write test can_update_min_hops_with_set_configuration_msg

* GH-692: set min hops value in db inside node_configurator/configurator.rs

* GH-692: only change value of min hops in neighborhood using SetConfigurationMessage

* GH-691: write more tests for setting min hops inside the node_configurator/configurator.rs

* GH-692: an attempt to remove the SetConsumingWalletMessage

* GH-692: remove SetConsumingWalletMessage from all the other places because it's a dead code

* GH-692: remove the SetConsumingWalletMessage completely

* GH-692: remove NewPasswordMessage

* GH-692: replace NewPasswordMessage with SetConfigurationMessage

* GH-692: add test for UpdateNewPassword inside configurator.rs

* GH-692: write more tests for the configurator.rs

* GH-692: improve test handle_set_configuration_handles_failure_on_min_hops_database_issue

* GH-692: change name of message to ConfigurationChangeMessage

* GH-691: implement calculate_db_patch_size()

* GH-692: add the fn to set_min_hops_and_db_patch_size()

* GH-692: set db_patch_size while setting min_hops

* GH-692: completely remove NewPasswordMessage

* GH-692: drive code for setting up patch size inside NeighborhoodMetadata

* GH-692: use db_patch_size or patch_size instead of min_hops while computing patch

* GH-692: remove redundant functions

* GH-692: formatting changes

* GH-692: ignore the stray test - ci/all.sh is passing

* GH-692: migrate min_hops test to a new file in multinode tests

* GH-692: attempt to write multinode test

* GH-692: why is this working? and not the previous version

* GH-692: improve multinode test

* GH-692: test drive the RouteQuery after the min hops has been changed

* GH-692: refactor the neighborhood/mod.rs file

* GH-692: min hops multinode test is passing

* GH-692: improve multinode test with better comments and refactoring

* GH-692: improve logging inside Neighborhood and Configurator

* GH-692: another formatting changes

* GH-692: allow overall connection stage to regress

* GH-692: regress stage of OverallConnectionStatus and send a msg to the UI

* GH-692: assert that the ui_gateway received a message for regression

* GH-692
: remove unnecessary warnings

* GH-692: test hardening

* GH-692: reorder msg inside recorder.rs

* GH-692: minor improvements in multinode tests

* GH-692: rename file

* GH-692: few fn renames

* GH-692: trigger actions

* GH-692: remove unused import from multinode test and increase timeout

* GH-692: regress ocs stage only if the initial stage is RouteFound

* GH-692: Review 1 (#334)

* GH-692: add the public constant DEFAULT_TRANSACT_TIMEOUT_MILLIS for the production code

* GH-692: add another constant for the tests

* GH-692: reduce total number of nodes for the min hops data routing test

* GH-692: change the comment for the multinode test

* GH-692: replace the constant for db_patch_size and use the different name everywhere

* GH-692: put code back for NewPasswordMessage

* GH-692: minor changes

* GH-692: add error logging when the UiSetConfigurationRequest causes an error

* GH-692: test fix and remove todo

* GH-692: fix test set_configuration_command_throws_err_for_invalid_arg

* GH-692: remove UpdateNewPassword variant

* GH-692: add better error msg in case the assert statement fails

* GH-692: fix using --gas-price and --min-hops during runtime without any value

* GH-692: required is actually not required

* GH-692: set_configurationify() changes min_values to 1 instead of 0

* GH-692: formatting

dnwiebe commented on node/src/proxy_server/mod.rs:
Wrong test name. Either correct or remove (for future compatibility).

Syther007 committed:
GH-651: Fixed System new name

dnwiebe commented on node/src/proxy_server/mod.rs:
Is the `zsh` on the end of this test name a mistake?

Syther007 committed:
GH-651: Removed zsh text

bertllll committed:
GH-751: Hot fix for wrong books keeping at transactions confirmation  (#349)

* GH-751: done

* GH-751: answer to the review

---------

Co-authored-by: Bert <Bert@Bert.com>

masqrauder committed:
GH-600: Add get_block_number (#193)

* Squashing:
GH-600: Add get_block_number
GH-600: Address PR feedback
GH-585: Experiment with adapting test for platform specific error codes
GH-585: Provide Windows platform specific error code and message
GH-600: Rebase master into branch

* GH-600: Handle expected batch, got single response

* Resolve rebase issues

* GH-600: Adapt to new Pokt message changes

* Use start_block in the event of an error

* GH-600: Return blockchain scan error to update max_block_count

* GH-600: Resolve blockchain scanning

* Starts Accountant
* Adds Receivables scan timeout (may become a configuration parameter in another card)
* Stores `max_block_count` in the config table when processing error messages providing a limit
* Adds Max Block Count to configuration response
* Increments CURRENT_SCHEMA_VERSION
* Adds schema migration for `max_block_count`
* Improves blockchain scan error handling and logging

* GH-600: Add test for scan time out

* GH-600: Revert scanner changes

* GH-600: Address review feedback

* GH-600: Address review 6 feedback

* GH-600: Use From traits to be concise

bertllll committed:
GH-672: payment adjuster infrastructure (#348)

* GH-554: rough infrastructure for PaymentAdjuster; almost get first round test running

* GH-672: simplifying GH-554, let's go on without the Neighborhood involved for the time being

* GH-672: PaymentAdjusterMock and the supertrait with ScannerEtensions implemented for a try

* GH-672-with-trait-impl-and-generics: perhaps going in the right direction

* GH-672-best-methodology-found: implementation works

* GH-672: first version of adjustment implemented

* GH-672: finished implementing the main test; not refactored yet

* GH-672: payment adjuster refactored; tuning logging; but wanna master features that are missing here

* GH-672: some logging integrated

* GH-672: fine debug log

* GH-672: interim commit

* GH-672: interim commit

* GH-672: another package of big changes (partly-private msgs; payment adjuster)

* GH-672: decent debug log for adjusted payables

* GH-672: broadening the adjustment capabilities with gas; rough framework

* GH-672: savepoint; before experimenting with early gas_limit computation

* GH-672: continuing in implementation of the check for gas availability.

* GH-672: some renaming and moving files

* GH-672: almost completed but there is a cenceptual error; maybe extra tests for BlockchainInterfaceClandestine should be made first

* GH-672: last details; adding tests for untested code from BlockchainInterfaceClandestine

* GH-672: almost done; some refactoring ahead

* GH-672: heading to auto-review

* GH-672: auto review findings

* GH-672: refactored scan scheduling code

* GH-672: fixing issues from the review; save point before the shift of responsibility of the estimated gas limit from BchI to Chain

* GH-672: before connecting the early fetched gas price and the actual gas price used for generating hashed transactions

* GH-672: logic redesigned + renaming

* GH-672: I believe the first review is answered by now

* GH-672: almost all the old code updated; will have to implement the transaction fee calculators

* GH-672: before serious changes with PayablePaymentsAgent starts

* GH-672-with-agent: saving when getting into troubles with testing

* GH-672-with-agent: going in a nice direction; interim commit

* GH-672-with-agent: approaching the fixture...this might work well; last vexing situations unhandled meaning there are like 8 failing tests at the moment

* GH-672-with-agent: could be called roughly complete; will go with refactoring and knocking off some todos left behind me

* GH-672-with-agent: interim commit

* GH-672-with-agent: hopefully the last changes before a review

* GH-672-with-agent: almost all what I found in my auto-review

* GH-672-with-agent: refactored the structure of PayablePaymentsSetupMsg

* GH-672-with-agent: formatting

* GH-672-wa-no-test-hack: recorder can also record and stop the system from PartalEq missing Actor messages; two of four tests repaired

* GH-672-wa-no-test-hack:  hurrah, managed to get out of the trap with an agent in Actor msgs even without having to implement Debug, PartialEq and Clone a trait object...this is definitely a lighter version

* GH-672-wa-no-test-hack: little modifications in comments

* GH-672: first batch of fixed items from the review 3

* GH-672: another batch of fixes

* GH-672: refactoring some complicated test concepts for actor_system_factory and adding one test for blockchain_bridge of that kind

* GH-672: more refactoring and little changes

* GH-672: fixing stuff araound blockchain_interface web3 and its initialization

* GH-672: savepoint

* GH-672: renaming

* GH-672: almost all things done; AgentDigest fully in

* GH-672: finished last few missing places

* GH-672: reworking comments and found a little mistake

* GH-672: match stop condition turned into easier one to use

* GH-672: protected payables implemented

* GH-672: until now the protected payables were implemented badly; improved design

* GH-672: compiling, but many failing tests

* GH-672: still fixing tests, but a lot of them passing now

* GH-672: halting hard with thoughts about implementation of the build_blockcahin_agent_method

* GH-672: blockchain interface onward and backwards move; messy rearrangement; wanna make a savepoint before trying cargo fix

* GH-672: all tests good; saving before cargo fix

* GH-672: roughly ready code; let's review now

* GH-672: renaming in accountant folders

* GH-672: reverting some renaming in favor of another branch taking care of this

* GH-672: interim commit

* GH-672: little fixes and adding more in to some null methods with todos

* GH-672: polishing; savepoint before moving get_transaction_receipt to plain_rpc

* GH-672: improved details

* GH-672: mostly done; just a few details left

* GH-672: abondoning enhancement of blockchain interface initializer...leaving as it is

* GH-672: unnoticable change for the augly low level tests in actor_system_factory

* GH-672: blockchain_bridge_shared_test_simplified

* GH-672: ready for review 5

* Squashing:
GH-600: Add get_block_number
GH-600: Address PR feedback
GH-585: Experiment with adapting test for platform specific error codes
GH-585: Provide Windows platform specific error code and message
GH-600: Rebase master into branch

* GH-600: Handle expected batch, got single response

* Resolve rebase issues

* GH-600: Adapt to new Pokt message changes

* Use start_block in the event of an error

* GH-600: Return blockchain scan error to update max_block_count

* GH-600: Resolve blockchain scanning

* Starts Accountant
* Adds Receivables scan timeout (may become a configuration parameter in another card)
* Stores `max_block_count` in the config table when processing error messages providing a limit
* Adds Max Block Count to configuration response
* Increments CURRENT_SCHEMA_VERSION
* Adds schema migration for `max_block_count`
* Improves blockchain scan error handling and logging

* GH-600: Add test for scan time out

* GH-600: Revert scanner changes

* GH-672: interim commit

* GH-672: finished review 5

* GH-672: review 6 answered

* GH-672: fixes in multinode tests

* GH-672: little cosmetics

* GH-672: renamings and moves in test code

* GH-672: rpc halpers into lower level interface

* Squashing:
GH-600: Add get_block_number
GH-600: Address PR feedback
GH-585: Experiment with adapting test for platform specific error codes
GH-585: Provide Windows platform specific error code and message
GH-600: Rebase master into branch

* GH-600: Handle expected batch, got single response

* Resolve rebase issues

* GH-600: Adapt to new Pokt message changes

* Use start_block in the event of an error

* GH-600: Return blockchain scan error to update max_block_count

* GH-600: Resolve blockchain scanning

* Starts Accountant
* Adds Receivables scan timeout (may become a configuration parameter in another card)
* Stores `max_block_count` in the config table when processing error messages providing a limit
* Adds Max Block Count to configuration response
* Increments CURRENT_SCHEMA_VERSION
* Adds schema migration for `max_block_count`
* Improves blockchain scan error handling and logging

* GH-600: Add test for scan time out

* GH-600: Revert scanner changes

* GH-600: Address review feedback

* GH-672: ending a big rearrangement; stable with all the tests again

* GH-672: more refactoring

* GH-672: missing import

* GH-672: review 7 answered

* GH-672: removed a not suitable comment

* GH-672: review 8 answered

* GH-672: lower interface renames

---------

Co-authored-by: Bert <Bert@Bert.com>
Co-authored-by: masqrauder <60554948+masqrauder@users.noreply.github.com>

utkarshg6 committed:
GH-716: Our StreamKeys are insecure (#336)

* GH-716: add UUID

* Revert "GH-716: add UUID"

This reverts commit f0a60402c6bf91f3f152aa7d63eafc1f534be7bb.

* GH-716: use the version of uuid that's been used for actix

* GH-716: add dbnavigator to gitignore

* GH-716: use UUIDs to make stream keys

* GH-716: remove parameters from the constructor of StreamKey

* GH-716: remove warnings

* GH-716: tests in stream_key.rs are passing

* GH-716: fix test handles_http_with_a_port

* GH-716: fix some more tests

* GH-716: make hash public for StreamKey for make_meaningless_stream_key()

* GH-716: small test fixes

* GH-716: fix test inbound_server_data_is_translated_to_cores_packages

* GH-716: make the test inbound_server_data_is_translated_to_cores_packages not pass easily

* GH-716: introduce fn make_meaningful_stream_key

* GH-716: improve test inbound_server_data_is_translated_to_cores_packages

* GH-716: remove todo

* GH-716: improve tests inside stream_key.rs

* GH-716: remove clippy warnings

* GH-716: use default() instead of new() in tests

* GH-716: fix test proxy_server_receives_tls_packet_other_than_handshake_from_dispatcher_then_sends_cores_package_to_hopper

* GH-716: trigger actions

* GH-716: an attempt to fix multinode tests

* GH-716: review changes

* GH-716: make the field name of StreamKey named 'hash' private

* GH-716: improve tests in file client_request_payload_factory.rs

* GH-716: share the helper functions in other crates

Syther007 committed:
Merge remote-tracking branch 'origin/master' into GH-651

# Conflicts:
#	node/src/proxy_server/mod.rs
#	node/src/test_utils/recorder.rs

Syther007 committed:
GH-651: Merged in Master
